--- hugo/internal/hugo_site.bzl
+++ hugo/internal/hugo_site.bzl
@@ -1,3 +1,5 @@
+load("//hugo:internal/hugo_theme.bzl", "HugoThemeInfo")
+
 def relative_path(src, dirname):
     """Given a src File and a directory it's under, return the relative path.
 
@@ -92,7 +94,7 @@ def _hugo_site_impl(ctx):
 
     # Copy the theme
     if ctx.attr.theme:
-        theme = ctx.attr.theme.hugo_theme
+        theme = ctx.attr.theme[HugoThemeInfo]
         hugo_args += ["--theme", theme.name]
         for i in theme.files.to_list():
             path_list = i.short_path.split("/")
@@ -205,9 +207,7 @@ hugo_site = rule(
         ),
         # Optionally set the base_url as a hugo argument
         "base_url": attr.string(),
-        "theme": attr.label(
-            providers = ["hugo_theme"],
-        ),
+        "theme": attr.label(),
         # Emit quietly
         "quiet": attr.bool(
             default = True,
@@ -244,7 +244,7 @@ def _hugo_serve_impl(ctx):
 
     if ctx.attr.quiet:
         hugo_args.append("--quiet")
-    if ctx.attr.quiet:
+    if ctx.attr.verbose:
         hugo_args.append("--verbose")
     if ctx.attr.disable_fast_render:
         hugo_args.append("--disableFastRender")